app/models/marc_fields/linked_related_works.rb

```diff
@@ -0,0 +1,94 @@
+##
+# A class to parse MARC related works (730 and others) and link them
+# to a search. Some subfields are not include in the search
+# but they do show up inside the link text, hence the logic in
+# the map/join methods.
+#
+class LinkedRelatedWorks < MarcField
+  LINK_CODES = %w[a d f k l m n o p r s t].freeze
+  TEXT_CODES = %w[h i x 3].freeze
+
+  def to_partial_path
+    'marc_fields/linked_related_works'
+  end
+
+  def tags
+    %w[700 710 711 720 730]
```

Comment:
yeah, it's complicated. `contributors_and_works_from_marc` parses those same tags ('700', '710', '711', '720') BUT only outputs results when `indicator2 == '2'`.  So, the LinkedRelatedWorks class filters out those with the `filter_out_indicator2` preprocessor.

I wasn't really sure how far to go in refactoring. For example, we could have a LinkedField class and then derive a LinkedRelatedWorks, LinkedIncludedWorks, and LinkedContributor and get rid of the `contributors_and_works_from_marc` function entirely. As it is in the PR now, I've only extracted the `LinkedRelatedWorks` (related_works) result from `contributors_and_works_from_marc`.